services/credentials-service.ts

```diff
@@ -0,0 +1,60 @@
+import * as path from "path";
+import * as crypto from "crypto";
+import { platform, EOL } from "os";
+
+export class CredentialsService implements ICredentialsService {
+	private static ENCRYPTION_ALGORITHM = "aes-256-ctr";
+	private static ENCRYPTION_KEY = "6Gsz97KBp293Q0r0ei4pix98V4PIhm2D";
+	private pathToWindowsCredentialsManager: string;
+
+	constructor(private $childProcess: IChildProcess,
+		private $hostInfo: IHostInfo) {
+		this.pathToWindowsCredentialsManager = path.join(__dirname, "..", "vendor", platform(), "CredentialsManager.exe");
+	}
+
+	public async setCredentials(key: string, credentials: ICredentials): Promise<ICredentials> {
+		if (this.$hostInfo.isWindows) {
+			await this.$childProcess.spawnFromEvent(this.pathToWindowsCredentialsManager, ["set", key, credentials.username, this.encrypt(credentials.password)], "close");
+			return credentials;
+		} else {
+			throw new Error(`Storing credentials is not supported on ${platform()} yet.`);
```

Comment:
I'm not sure what you mean here. In all cases the hostname, port and USE_PROXY values are persisted in a file inside a special folder (known as profileDir in CLI). The options from config.json are deleted and they are no longer supported. Only the credentials are stored in a special location.